test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageBuilderTest.cs

```diff
@@ -2485,6 +2487,73 @@ var contentTypesReader
             }
         }
 
+        [Fact]
+        public void Icon_IconMaxFileSizeExceeded_ThrowsException()
+        {
+            TestIconPackaging(
+                new IconTestSourceDirectory("icon.jpg", "icon.jpg", PackageBuilder.MaxIconFileSize + 1),
+                "The icon file size must not exceed 1 megabyte.");
+        }
+
+        [Fact]
+        public void Icon_IconFileEntryNotFound_ThrowsException()
+        {
+            TestIconPackaging(
+                new IconTestSourceDirectory("icon.jpg", "icono.jpg", 100),
+                "No icon <file> element found for <icon> in the nuspec.");
+        }
+
+        [Fact]
+        public void Icon_HappyPath_Suceed()
+        {
+            TestIconPackaging(
+                new IconTestSourceDirectory("icon.jpg", "icon.jpg", 400),
+                string.Empty);
+        }
+
+        [Fact(Skip="Need to solve https://github.com/NuGet/Home/issues/6941 to run this test case")]
+        public void Icon_MultipleIconFilesResolved_ThrowsException()
+        {
+            var fileList = new List<Tuple<string, int>>();
+            var fileEntriesList = new List<string>();
+            var dirSep = Path.DirectorySeparatorChar;
+
+            fileList.Add(Tuple.Create(string.Format("folder1{0}icon.jpg", dirSep), 2));
+            fileList.Add(Tuple.Create(string.Format("folder1{0}dummy.txt", dirSep), 2));
+            fileList.Add(Tuple.Create(string.Format("folder2{0}icon.jpg", dirSep), 2));
+            fileList.Add(Tuple.Create(string.Format("folder2{0}file.txt", dirSep), 2));
+
+            fileEntriesList.Add("folder1\\*");
+            fileEntriesList.Add("folder2\\*");
+
+            TestIconPackaging(
+                new IconTestSourceDirectory("icon.jpg", fileList, fileEntriesList),
+                "Multiple files resolved as the embedded icon.");
+        }
+
+
+
+        private void TestIconPackaging(IconTestSourceDirectory sourceDir, string exceptionMessage)
+        {
+            using (sourceDir)
+            using (var nuspecStream = File.OpenRead(sourceDir.NuspecPath))
+            using (var outputNuPkgStream = new MemoryStream())
+            {
+                PackageBuilder pkgBuilder = new PackageBuilder(nuspecStream, sourceDir.BaseDir);
+
+                if (!string.IsNullOrEmpty(exceptionMessage))
```

Comment:
```suggestion
                if (exceptionMessage != null)
```

I think only allowing null is more precise. The type system allows us to differentiate between "is not specified" and "is specified with an empty value", so I think it's more clear to use that.

Reply:
Agree. Makes the code more readable.